File: MyList.TestHelpers/ShoppingListBuilder.cs
using MyList.Entity;
using System;
using System.Collections.Generic;

namespace MyList.TestHelpers
{
    public class ShoppingListBuilder
    {
        private Guid _id = Guid.NewGuid();
        private string _name = "defaultList";
        private List<ShoppingListItem> _items = new List<ShoppingListItem>() 
        { 
            ShoppingListItemBuilder.Create().Build()
        };

        public ShoppingListBuilder()
        { }

        public static ShoppingListBuilder Create()
        {
            return new ShoppingListBuilder();
        }

        public ShoppingListBuilder WithId(Guid id)
        {
            _id = id;
            return this;
        }

        public ShoppingListBuilder WithName(string name)
        {
            _name = name;
            return this;
        }

        public ShoppingListBuilder WithItems(List<ShoppingListItem> items)
        {
            _items = items;
            return this;
        }

        public ShoppingListBuilder WithoutItems()
        {
            _items = null;
            return this;
        }

        public ShoppingList Build()
        {
            return new ShoppingList()
            {
                Id = _id,
                Name = _name,
                Items = _items
            };
        }
    }
}

File: MyList.Web/src/app/test/builders/shopping-list.builder.ts
import { ShoppingListItem } from 'src/app/models/shopping-list-item';
import { ShoppingList } from './../../models/shopping-list';
import { ShoppingListItemBuilder } from './shopping-list-item.builder';

export class ShoppingListBuilder {
  private _id: string = "1";
  private _name: string = "defaultName";
  private _createdDate: Date = new Date();
  private _items: ShoppingListItem[] = [ ShoppingListItemBuilder.create().build() ];

  constructor() {}

  static create = () => new ShoppingListBuilder();

  withId(id: string): ShoppingListBuilder {
    this._id = id;
    return this;
  }

  withName(name: string): ShoppingListBuilder {
    this._name = name;
    return this;
  }

  withCreatedDate(date: Date): ShoppingListBuilder {
    this._createdDate = date;
    return this;
  }

  withItems(items: ShoppingListItem[]): ShoppingListBuilder {
    this._items = items;
    return this;
  }

  withoutItems(): ShoppingListBuilder {
    this._items = null;
    return this;
  }

  build(): ShoppingList {
    return {
      id: this._id,
      name: this._name,
      createdDate: this._createdDate,
      items: this._items
    } as ShoppingList
  }
}
